best_practices/production_deployments/plugins/WITHOUT_COMPILERS.md

```diff
@@ -1,5 +1,32 @@
 # Pre-compiling native extensions for Sensu Plugins gem dependencies
 
+## What is this about
+
+Basically many of the plugins require the use of a compiler (typically c) on install time. There are good reasons to avoid compilers on systems and bad reasons to do so.
+
+### Good reasons
+
+1. Reduce spin-up time, if you have instances in some sort of scaling group and you want to reduce the amount of time it takes for a machine to come up with monitoring. An alternative to this is to bake the gem installs into your image.
+1. Compilers are dev tools, they have no business being in a production environment.
+
+### Bad reasons
+
+1. Security: I have heard the argument many times that somehow having a compiler makes you less secure. This statement is often made by security teams that have no idea what they are talking about. If the attacker has enough access to a system to use a compiler then they would opt for either compiling on their machine and dropping the executable on the client or leverage the tools that are already on the box. The 2 most common targets at this poing become the various `shell` and `interpretors` such as `python` which is almost always guaranteed on `*nix` or `ruby` which currently is guaranteed on any sensu install, etc. This is **Security Theater** and does not really reduce any **real** attack vectors. This creates extra complexity in deployment without a good reason.
```

Comment:
Minor typo: poing -> point

Reply:
shell and interpreters don't need to be code highlighted. Also typo of interpreters.

Reply:
Love the strong opinion 😄

Reply:
no syntax highlight on *nix either, but ruby/python makes sense since you're talking about the compiler.